mesonbuild/environment.py

```diff
@@ -1109,61 +1110,38 @@ def parse_datafile(self, filename):
                 except Exception:
                     raise EnvironmentException('Malformed value in cross file variable %s.' % entry)
 
-                if self.ok_type(res):
-                    self.config[s][entry] = res
+                if self._ok_type(res):
+                    self[s][entry] = res
                 elif isinstance(res, list):
                     for i in res:
-                        if not self.ok_type(i):
+                        if not self._ok_type(i):
                             raise EnvironmentException('Malformed value in cross file variable %s.' % entry)
-                    self.config[s][entry] = res
+                    self[s][entry] = res
                 else:
                     raise EnvironmentException('Malformed value in cross file variable %s.' % entry)
 
-    def has_host(self):
-        return 'host_machine' in self.config
+    @classmethod
+    def _ok_type(cls, i):
+        return isinstance(i, (str, int, bool))
```

Comment:
at this point this is used in one function, you could either just make it local, or at the very least make it a `@staticmethod` instead of a `@classmethod`

Reply:
inlined in followup commit.